relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py

```diff
@@ -214,18 +215,22 @@ def check_url():
         fail('An error occurred while checking access to %s:\n%s' % (URL, e))
 
 
-"""
+'''
    From a complete url, try to download a file. The destination path will be determined by the complete url
    after removing the prefix designing the repo url defined in the conf file.
    Ex: completeUrl = http://download.rudder.io/plugins/./5.0/windows/release/SHA512SUMS
            repoUrl = http://download.rudder.io/plugins
         => fileDst = /tmp/rpkg/./5.0/windows/release/SHA512SUMS
-"""
+'''
 
 
 def download(completeUrl, dst=''):
     if dst == '':
-        fileDst = FOLDER_PATH + '/' + completeUrl.replace(URL + '/' + RUDDER_MINOR + '/', '')
+        # only take file name to avoid reaching max file name length
+        fileName = completeUrl.rsplit('/', 1)[-1]
+        # escape but keep tildes
+        fileName = urllib.parse.quote(fileName, '~')
+        fileDst = FOLDER_PATH + '/' + fileName
```

Comment:
Looks good, can you force the `rpkg.index` download path in `rudderPkg.py` using something like:
`utils.download(utils.URL + '/' + 'rpkg.index', utils.INDEX_PATH)`